atlas/src/shared/components/Carousel/Carousel.tsx

```diff
@@ -1,94 +1,70 @@
-import NavButton from '../NavButton'
+import React, { useState, useEffect, useRef } from 'react'
+import { GliderProps, BreakPoint, GliderMethods } from 'react-glider'
 
-export type CarouselProps = {
-  containerCss?: SerializedStyles
-  leftControlCss?: SerializedStyles
-  rightControlCss?: SerializedStyles
-  disableControls?: boolean
-  onScroll?: (direction: 'left' | 'right') => void
-} & CarouselStyleProps
+import { Container, StyledGlider, Arrow } from './Carousel.style'
 
-const Carousel: React.FC<CarouselProps> = ({
-  children,
-  containerCss,
-  leftControlCss,
-  rightControlCss,
-  disableControls = false,
-  onScroll = () => {},
-}) => {
-  const [scroll, setScroll] = useSpring(() => ({
-    transform: `translateX(0px)`,
-  }))
-  const [carouselOffset, setCarouselOffset] = useState(0)
-  const { width: containerWidth = 0, ref: containerRef } = useResizeObserver<HTMLDivElement>()
-  const { width: childrenWidth = 0, ref: childrenContainerRef } = useResizeObserver<HTMLDivElement>()
+import 'glider-js/glider.min.css'
 
-  const styles = useCSS({})
+type CarouselProps = {
+  trackPadding?: string
+} & GliderProps
 
-  const maxScrollOffset = childrenWidth - containerWidth
+type TrackProps = {
+  className?: string
+  padding?: string
+}
+const Track: React.FC<TrackProps> = ({ className = '', ...props }) => (
+  <div className={`glider-track ${className}`} {...props} />
+)
 
-  const showLeftControl = !disableControls && carouselOffset > 0
-  const showRightControl = !disableControls && carouselOffset < maxScrollOffset
+const RightArrow = <Arrow name="chevron-right" />
+const LeftArrow = <Arrow name="chevron-left" />
 
-  const handleScroll = (direction: 'left' | 'right') => {
-    if (containerWidth == null) {
-      return
-    }
-    let scrollAmount
-    switch (direction) {
-      case 'left': {
-        // Prevent overscroll on the left
-        const newOffset = carouselOffset - containerWidth
-        scrollAmount = newOffset < 0 ? 0 : newOffset
-        onScroll('left')
-        break
-      }
-      case 'right': {
-        // Prevent overscroll on the right
-        const newOffset = carouselOffset + containerWidth
-        scrollAmount = newOffset > maxScrollOffset ? maxScrollOffset : newOffset
-        onScroll('right')
-        break
-      }
+const Carousel: React.FC<CarouselProps> = ({ children, trackPadding = '0', className, ...gliderProps }) => {
+  // Using any because the GliderMethods type only has methods and I need the full instance
+  const gliderRef = React.useRef<any>()
+  const [arrows, setArrows] = React.useState<
+    { prev: string | HTMLButtonElement; next: string | HTMLButtonElement } | undefined
+  >(undefined)
+
+  React.useLayoutEffect(() => {
+    if (gliderRef.current) {
+      const glider = gliderRef.current.ele
+      const prevArrow = glider.previousSibling
+      const nextArrow = glider.nextSibling
+      const INSTANCE_KEY = Math.round(Math.random() * 1000)
```

Comment:
If I understand correctly, this has a 1/1000 chance of being a duplicate, hence potentially breaking some other carousel in the same view?

Reply:
Correct. Being such a small chance `#carousels on the same page / 1000` I didn't bother to add a real library like `uuid` to take care of this, but if you think this might be an issue, I'll add it.

Reply:
We currently have 3 carousels in the home view, so statistically, every 333rd visitor will encounter a bug, that's not a very good score

Reply:
I just realised that an `ISTANCE_KEY` is only useful if the `arrows` object holds strings, but since we're passing the elements directly (and those are tied to a specific instance) there was no need to add the class and have the `INSTANCE_KEY`